libraries/c_sdk/standard/mqtt/src/iot_mqtt_operation.c

```diff
@@ -498,8 +499,19 @@ bool _IotMqtt_DecrementOperationReferences( _mqttOperation_t * pOperation,
         EMPTY_ELSE_MARKER;
     }
 
-    /* Decrement job reference count. */
-    if( taskPoolStatus == IOT_TASKPOOL_SUCCESS )
+    /*
+     * The job reference for the given operation needs to be deceremented in the following cases:
+     * 1. For CONNECT operation , if the taskpool status is IOT_TASKPOOL_SUCCESS.
+     *    If the CONNECT opeartion is still executing in the taskpool, then operation reference should not be decremented.
+     *    Shim Implementation is using taskpool for connect operation so this logic is updated.
```

Comment:
connect, does it need to be in uppercase for consistency?

Reply:
Yeah , this is to maintain the consistency.